packages/core/src/auth.ts

```diff
@@ -108,11 +108,11 @@ export async function createUserWithEmailAndPassword(
 export async function signInWithPhoneNumber(
   ui: FirebaseUIConfiguration,
   phoneNumber: string,
-  recaptchaVerifier: RecaptchaVerifier
+  appVerifier: ApplicationVerifier
 ): Promise<ConfirmationResult> {
   try {
     ui.setState("pending");
-    return await _signInWithPhoneNumber(ui.auth, phoneNumber, recaptchaVerifier);
+    return await _signInWithPhoneNumber(ui.auth, phoneNumber, appVerifier);
```

Comment:
```suggestion
    return _signInWithPhoneNumber(ui.auth, phoneNumber, appVerifier);
```

Reply:
The await is needed here to ensure a thrown error is bubbled to the catch, rather than being consumed in the promise chain